Relays+Infallible

mashe opened the pull request:
[original issue](https://github.com/ReactiveX/RxSwift/issues/2429)

freak4pc commented on RxRelay/Relays+Infallible.swift:
```diff
@@ -0,0 +1,34 @@
+    }
+}
+
+public extension PublishRelay {
```
Definitely needs some test coverage :)

freak4pc replied:
(basically just something that compares the results of the relay to the infallible is "good enough" here).

freak4pc reviewed (changes requested):
Lets get some tests on this and we can merge it in.
Thanks!

freak4pc commented:
@mashe got the time to take this to the finishing line? 🙏

rtharston commented:
This should just replace the relay specific parts of #2401.
Basically, just replace `asInfallible(onErrorFallbackTo: .empty())` in the methods added in that PR with `Infallible(self.asObservable())`

TTOzzi commented:
It already has test cases in [Infallible+Tests.swift](https://github.com/ReactiveX/RxSwift/blob/main/Tests/RxSwiftTests/Infallible%2BTests.swift#L104) :thinking:

mashe commented:
Thank you, everyone for keeping eye on the PR. 
@freak4pc, I have followed @rtharston suggestion, so I reverted initial commit and updated existing `.asInfallible()` functions for Relays. I believe additional tests are not required, since @TTOzzi noted that those exist already.

rtharston commented:
@freak4pc, any update on this?

freak4pc commented:
Cleaning up the repo and closing old PRs. If this is still something you're interested in pursuing, feel free to comment and we can re-open and discuss. Thanks!
